tests/common/plugins/sanity_check/checks.py

```diff
@@ -372,98 +426,63 @@ def _check(*args, **kwargs):
-        except AssertionError:
-            results['failed'] = True
-            results['failed_reason'] = 'Packet from standby upper ToR received'
-            return results
-
-        upper_tor_host.shell("ip neigh flush all")
-        lower_tor_host.shell("ip neigh flush all")
-        testutils.send_packet(ptfadapter, ptf_port_index, ptf_arp_pkt)
-
-        upper_tor_arp_table = upper_tor_host.switch_arptable()['ansible_facts']['arptable']['v4']
-        lower_tor_arp_table = lower_tor_host.switch_arptable()['ansible_facts']['arptable']['v4']
-        if ptf_arp_tgt_ip not in upper_tor_arp_table:
-            results['failed'] = True
-            results['failed_reason'] = 'Packet from PTF not received on standby upper ToR'
-            return results
-
-        if ptf_arp_tgt_ip not in lower_tor_arp_table:
-            results['failed'] = True
-            results['failed_reason'] = 'Packet from PTF not received on active lower ToR'
-            return results
-
-        logger.info('Finished mux simulator check')
-        upper_tor_host.shell("ip neigh flush all")
-        lower_tor_host.shell("ip neigh flush all")
+            # Stop linkmgrd to prevent it from switching over ports
+            lower_tor_host.shell('systemctl stop mux')
+            upper_tor_host.shell('systemctl stop mux')
+
+            toggle_simulator_port_to_upper_tor(tor_mux_intf)
+            if check_simulator_read_side(tor_mux_intf) != 1:
+                failed = True
+                reason = 'Unable to switch active link to upper ToR'
+
+            if not failed:
+                failed, reason = mux_sim_check_downstream(upper_tor_host, lower_tor_host,
+                    ptfadapter, upper_tor_ping_tgt_ip, lower_tor_ping_tgt_ip,
+                    upper_tor_exp_pkt, lower_tor_exp_pkt, ptf_port_index, tor_mux_intf)
+
+            
+            if not failed:
+                failed, reason = mux_sim_check_upstream(upper_tor_host, lower_tor_host,
+                    ptfadapter, ptf_arp_tgt_ip, ptf_arp_pkt, ptf_port_index)
+
+            # Repeat all tests with lower ToR active
+            if not failed:
+                toggle_simulator_port_to_lower_tor(tor_mux_intf)
+                if check_simulator_read_side(tor_mux_intf) != 2:
+                    failed = True
+                    reason = 'Unable to switch active link to lower ToR'
+
+            if not failed:
+                failed, reason = mux_sim_check_downstream(lower_tor_host, upper_tor_host,
+                    ptfadapter, lower_tor_ping_tgt_ip, upper_tor_ping_tgt_ip,
+                    lower_tor_exp_pkt, upper_tor_exp_pkt, ptf_port_index, tor_mux_intf)
+
+            if not failed:
+                failed, reason = mux_sim_check_upstream(upper_tor_host, lower_tor_host,
+                    ptfadapter, ptf_arp_tgt_ip, ptf_arp_pkt, ptf_port_index)
+
+            logger.info('Finished mux simulator check')
+        finally:
+            upper_tor_host.shell("ip neigh flush all")
```

Comment:
I think this command can also be added to the list of `cmds` to be executed below.